packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/amplify-meta-updaters.ts

```diff
@@ -106,10 +106,24 @@ export const getPostUpdateAuthMetaUpdater = (context: any) => async (resourceNam
   return resourceName;
 };
 
-function getFrontendConfig(authParameters: AuthParameters) {
+export function getFrontendConfig(authParameters: AuthParameters) {
   const loginMechanisms: string[] = [];
   loginMechanisms.push(...(authParameters?.aliasAttributes || []).map((att: string) => att.toUpperCase()));
 
+  // backwards compatibility
+  if (authParameters?.usernameAttributes) {
+    const usernameAttributes = authParameters?.usernameAttributes[0];
```

Comment:
can we just do a plain split and process the result and remove regex magic? that would be a much simpler code path IMHO, does this code guarantees to NOT to end up duplicate entries somehow? Perhaps using a `Set` instead of pure array would protect against that

Reply:
updated